gst/tensor_reposrc/tensor_reposrc.c

```diff
@@ -176,6 +177,13 @@ gst_tensor_reposrc_getcaps (GstBaseSrc * src, GstCaps * filter)
       cap = gst_caps_new_any ();
   }
 
+  st = gst_caps_get_structure (cap, 0);
```

Comment:
cap may have multiple instances of structure (e.g.., other/tensor, other/tensors at the same time).
rather than checking all structures, I'd recommend to do "cap intersect" with "GST_TENSORS_CAP_DEFAULT + GST_TENSOR_CAP_DEFAULT". Then, you can return the resulting cap. (and print error if the result it NULL)

Reply:
done. thanks.